pymatgen/core/structure.py

```diff
@@ -897,13 +907,18 @@ def __init__(
             if site_properties:
                 prop = {key: val[idx] for key, val in site_properties.items()}
 
+            label = None
+            if labels:
```

Comment:
This can be a ternary.
```suggestion
label = labels[idx] if labels else None
```